submissions/kt-std/friends-app/js/script.js

```diff
@@ -0,0 +1,447 @@
+    option.tabIndex = -1;
+    option.removeAttribute("aria-selected");
+}
+
+function getSelectOption(selectText) {
+    return Array.from(OPTIONS_CONTAINER.children).filter(
+        (option) => option.textContent === selectText)[0];
+}
+
+function higlightSelectedOption(itemToSelect) {
+    itemToSelect.tabIndex = 0;
+    setTimeout(() => {
+        itemToSelect.focus();
+    }, 100);
+    itemToSelect.setAttribute("aria-selected", "true");
+    updateSelectText(itemToSelect.textContent);
+}
+
+function focusOnItem(buttonPressed) {
+    const selectedItem = document.activeElement;
+    if (buttonPressed === "ArrowUp") {
+        switch (true) {
+            case selectedItem.previousElementSibling &&
+                isSelectOption(selectedItem):
+                higlightSelectedOption(selectedItem.previousElementSibling);
+                break;
+            case !selectedItem.previousElementSibling:
+                higlightSelectedOption(OPTIONS_CONTAINER.firstElementChild);
+                break;
+            case !isSelectOption(selectedItem):
+                higlightSelectedOption(OPTIONS_CONTAINER.lastElementChild);
+                break;
+        }
+    } else if (buttonPressed === "ArrowDown") {
+        switch (true) {
+            case selectedItem.nextElementSibling &&
+                isSelectOption(selectedItem):
+                higlightSelectedOption(selectedItem.nextElementSibling);
+                break;
+            case !selectedItem.nextElementSibling:
+                higlightSelectedOption(OPTIONS_CONTAINER.lastElementChild);
+                break;
+            case !isSelectOption(selectedItem):
+                higlightSelectedOption(OPTIONS_CONTAINER.firstElementChild);
+        }
+    }
+    resetOptionTabindex(selectedItem);
+}
+
+function isSelectOption(selectedItem) {
+    return (
+        selectedItem.classList.contains("select__list") ||
+        selectedItem.classList.contains("list__item")
+    );
+}
+
+function observeOptionsListVisibility() {
+    const options = { attributes: true, attributesFilter: ["classList"] },
+        callback = function (mutationsList, observer) {
+            for (const mutation of mutationsList) {
+                if (mutation.type === "attributes") {
+                    if (
+                        mutation.target.classList.contains("visible") &&
+                        !SELECT_CONTAINER.getAttribute("aria-expanded")
+                    ) {
+                        SELECT_CONTAINER.setAttribute("aria-expanded", "true");
+                    }
+                    if (
+                        !mutation.target.classList.contains("visible") &&
+                        SELECT_CONTAINER.getAttribute("aria-expanded")
+                    ) {
+                        SELECT_CONTAINER.removeAttribute("aria-expanded");
+                    }
+                }
+            }
+        };
+    const observer = new MutationObserver(callback);
```

Comment:
You also can do this logic in the event listener where you open your filter.

But `MutationObserver` is fine.